source/diagnostic.h

```diff
@@ -30,8 +30,12 @@ class DiagnosticStream {
  public:
   DiagnosticStream(spv_position_t position,
```

Comment:
Given that there are so many times you call this constructor with an empty string value for the disassembled_instruction member, I would have probably made a second constructor.

Reply:
I'm on the fence for this one. It makes it a lot clearer where we don't have a disassembled_instruction and, hopefully, would make it harder to add a missing one in the future. It's also possible, some of the fields I have as blank could have values, like bad header and a couple others.

Reply:
> would make it harder to add a missing one in the future.  

That's a good argument.